### PR TITLE
Fix - Refactor cURL error handling to preserve response data

### LINKED ISSUE #32416: Improve logging on Manager
## Description

The current logging in the *indexer-connector* module is inconsistent and often lacks clear context. This makes it difficult to quickly understand what went wrong or what the system is doing. We need to enhance the logging so that *every* log emitted by the module is clear, actionable, and consistent.

This has been noticed during the support provided on

- https://github.com/wazuh/wazuh/issues/32044

## Definition of Done  
- [x] All log statements in the module provide enough context about the error and determinate the root cause.  
- [x] Tests or sample runs confirm that logs appear and are clear in scenarios (success, error, edge cases)

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
Closes https://github.com/wazuh/wazuh/issues/32416

# Summary

This change refactors the cURL execution logic to improve error handling and diagnostics. Fixes errors that were found during the E2E testing of #75

### CODE REVIEW

File: src/curlSingleHandler.hpp [hunk 1]
```diff
@@ -50,24 +50,46 @@ class cURLSingleHandler final : public ICURLHandler
...
             }
-            throw Curl::CurlException(curl_easy_strerror(resPerform), NOT_USED);
+            else if (responseCode >= 500)
+            {
+                errorMsg = "Server error";
+            }
+            else
+            {
+                errorMsg = "HTTP error";
+            }
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/curlSingleHandler.hpp]: I agree with the bot here. 

external if only accepts status code from 400 and above, the first if [400, 500) and the else if [500, inf), what was the purpose of the last else?

File: test_tool/actions.hpp [hunk 2]
```diff
@@ -70,13 +70,14 @@ class DownloadAction final : public IAction
...
-                                   {
-                                       std::cerr << msg << ": " << responseCode << std::endl;
-                                       throw std::runtime_error(msg);
-                                   },
-                                   .outputFile = m_outputFile},
+            PostRequestParameters {
+                .onError =
+                    [](const std::string& msg, const long responseCode, const std::string& resultBody)
+                {
+                    std::cerr << msg << ": " << responseCode << std::endl;
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply test_tool/actions.hpp]: I guess this was added to fix some compilation error, since now the error lambda receives a new parameter. Is that correct because it is not used. 

Anyway it does not bother me, it's the testtool.

File: src/curlSingleHandler.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/curlSingleHandler.hpp]: Fixed!

File: test_tool/actions.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply test_tool/actions.hpp]: Fixed!

File: test_tool/actions.hpp [hunk 3]
```diff
@@ -169,19 +171,19 @@ class PostAction final : public IAction
...
-                                     ConfigurationParameters {.timeout = m_timeout});
+        HTTPRequest::instance().post(
+            RequestParameters {
+                .url = HttpURL(m_url), .secureCommunication = m_secureCommunication, .httpHeaders = m_headers},
+            PostRequestParameters {
+                .onSuccess = [](const std::string& msg) { std::cout << msg << std::endl; },
+                .onError =
+                    [](const std::string& msg, const long responseCode, const std::string& resultBody)
+                {
+                    std::cerr << msg << ": " << responseCode << std::endl;
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply test_tool/actions.hpp]: Fixed!

File: test_tool/actions.hpp [hunk 4]
```diff
@@ -229,13 +231,14 @@ class PutAction final : public IAction
...
-                                   {
-                                       std::cerr << msg << ": " << responseCode << std::endl;
-                                       throw std::runtime_error(msg);
-                                   }},
+            PostRequestParameters {
+                .onSuccess = [](const std::string& msg) { std::cout << msg << std::endl; },
+                .onError =
+                    [](const std::string& msg, const long responseCode, const std::string& resultBody)
+                {
+                    std::cerr << msg << ": " << responseCode << std::endl;
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply test_tool/actions.hpp]: Fixed!

File: test_tool/actions.hpp [hunk 5]
```diff
@@ -336,13 +340,14 @@ class DeleteAction final : public IAction
...
-                                   {
-                                       std::cerr << msg << ": " << responseCode << std::endl;
-                                       throw std::runtime_error(msg);
-                                   }},
+            PostRequestParameters {
+                .onSuccess = [](const std::string& msg) { std::cout << msg << std::endl; },
+                .onError =
+                    [](const std::string& msg, const long responseCode, const std::string& responseBody)
+                {
+                    std::cerr << msg << ": " << responseCode << std::endl;
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply test_tool/actions.hpp]: Fixed!

File: src/curlMultiHandler.hpp [hunk 6]
```diff
@@ -111,11 +115,17 @@ class cURLMultiHandler final : public ICURLHandler
...
                 {
                     auto errorCode = multiHandleMessages->data.result;
+
+                    // Check for cURL-level errors (network, DNS, timeout, etc.)
                     if (errorCode != CURLE_OK)
                     {
-                        throw Curl::CurlException("cURLMultiHandler::execute() failed: " +
-                                                      std::string(curl_easy_strerror(errorCode)),
-                                                  errorCode);
+                        throw Curl::CurlException(curl_easy_strerror(errorCode), -1);
```
Comment (MarcelKemp) [inline src/curlMultiHandler.hpp]: Why don't we use `NOT_USED`?
```suggestion
                        throw Curl::CurlException(curl_easy_strerror(errorCode), NOT_USED); 
```

**Review by MarcelKemp — APPROVED** [review]

File: src/curlMultiHandler.hpp (hunk 6 above)
Comment (GabrielEValenzuela) [reply src/curlMultiHandler.hpp]: Solved!
